kernel/kernel-api/src/main/java/io/delta/kernel/internal/CreateTableTransactionBuilderImpl.java

```diff
@@ -0,0 +1,121 @@
+   * Delta-Spark, for historical reasons the number of retries is really high (10m). We are starting
+   * with a lower number by default for now. If this is not sufficient we can update it.
+   */
+  private int maxRetries = 200;
+
+  public CreateTableTransactionBuilderImpl(String tablePath, StructType schema, String engineInfo) {
+    this.unresolvedPath = requireNonNull(tablePath, "tablePath is null");
+    this.schema = requireNonNull(schema, "schema is null");
+    this.engineInfo = requireNonNull(engineInfo, "engineInfo is null");
+  }
+
+  @Override
+  public CreateTableTransactionBuilder withTableProperties(Map<String, String> properties) {
+    requireNonNull(properties, "properties cannot be null");
+    this.tableProperties =
+        Optional.of(
+            Collections.unmodifiableMap(
+                TableConfig.validateAndNormalizeDeltaProperties(properties)));
+    return this;
+  }
+
+  @Override
+  public CreateTableTransactionBuilder withDataLayoutSpec(DataLayoutSpec spec) {
+    requireNonNull(spec, "spec cannot be null");
+    this.dataLayoutSpec = Optional.of(spec);
+    return this;
+  }
+
+  @Override
+  public CreateTableTransactionBuilder withMaxRetries(int maxRetries) {
+    checkArgument(maxRetries >= 0, "maxRetries must be >= 0");
+    this.maxRetries = maxRetries;
+    return this;
+  }
+
+  @Override
+  public Transaction build(Engine engine) {
+    requireNonNull(engine, "engine cannot be null");
+    String resolvedPath = resolvePath(engine, unresolvedPath);
+
+    // Extract partition and clustering columns from the data layout spec
+    Optional<List<String>> partitionColumns =
+        dataLayoutSpec
+            .filter(DataLayoutSpec::hasPartitioning)
+            .map(
+                spec ->
+                    spec.getPartitionColumns().stream()
+                        .map(col -> col.getNames()[0])
+                        .collect(Collectors.toList()));
+    Optional<List<Column>> clusteringColumns =
+        dataLayoutSpec
+            .filter(DataLayoutSpec::hasClustering)
+            .map(DataLayoutSpec::getClusteringColumns);
+
+    TransactionMetadataFactory.Output txnMetadata =
+        TransactionMetadataFactory.buildCreateTableMetadata(
+            resolvedPath,
+            schema,
+            tableProperties.orElse(Collections.emptyMap()),
+            partitionColumns,
+            clusteringColumns);
+
+    Path dataPath = new Path(resolvedPath);
+    return new TransactionImpl(
+        true, // isCreateOrReplace
+        dataPath,
+        Optional.empty(), // no existing snapshot for create table
+        engineInfo,
+        Operation.CREATE_TABLE,
+        txnMetadata.newProtocol,
+        txnMetadata.newMetadata,
+        Optional.empty(), // no setTransaction for create table
+        txnMetadata.physicalNewClusteringColumns,
+        maxRetries,
+        0, // logCompactionInterval - using default for create table
```

Comment:
can this be a private static final int, just like with `maxRetries` ?

Reply:
and btw should there be a `withLogCompactionInterval` API? or are we moving to a config?